### PR TITLE
[bitnami/thanos] added psp for thanos querier

### PR DESCRIPTION
**Pull request by ahmednasir91**
**Description of the change**

Added PodSecurityPolicy for thanos querier.

**Benefits**

Allow the users to create PSP for thanos querier.

**Checklist**
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/thanos/templates/querier/psp.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - max: 1001
+      min: 1001
+    rule: MustRunAs
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+  - secret
+---
```
Comment (marcosbc) [inline bitnami/thanos/templates/querier/psp.yaml:22]: We usually separate ClusterRole and ClusterRoleBinding into different files (`psp-clusterrole.yaml` and `psp-clusterrolebinding.yaml` respectively). Could you make this change?

File: bitnami/thanos/templates/querier/psp.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{{- if and .Values.querier.enabled .Values.querier.psp.create -}}
```
Comment (marcosbc) [inline bitnami/thanos/templates/querier/psp.yaml:1]: The conditions to enable this for other charts is usually shown like this:

```
{{- if and .Values.rbac.create .Values.rbac.pspEnabled }}
```

Is there any reason for following a different format (via .Values.[component].psp.create)? For instance, have a look at [Prometheus](https://github.com/bitnami/charts/tree/master/bitnami/kube-prometheus/templates/prometheus).

**Review by marcosbc — COMMENTED** [review]
Thanks for the contribution! I've added a few comments, please have a look at them whenever you have a chance.

File: bitnami/thanos/templates/querier/psp.yaml (hunk 2 above)
Comment (ahmednasir91) [reply bitnami/thanos/templates/querier/psp.yaml:1]: @marcosbc the reason being in prometheus is just one component, here we have multiple components `querier`, `ruler` etc. Followed the same approach being done for PDB

https://github.com/bitnami/charts/blob/master/bitnami/thanos/templates/querier/pdb.yaml#L1

Comment (ahmednasir91) [reply bitnami/thanos/templates/querier/psp.yaml]: I've changed the condition for psp to be `pspEnabled` to make it consistent.

File: bitnami/thanos/templates/querier/psp.yaml (hunk 1 above)
Comment (ahmednasir91) [reply bitnami/thanos/templates/querier/psp.yaml]: Done

File: bitnami/thanos/templates/querier/psp.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  kind: ClusterRole
+  name: {{ include "thanos.fullname" . }}-querier
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+# Authorize specific service accounts:
+- kind: ServiceAccount
+  name: {{ include "thanos.serviceaccount.name" (dict "component" "querier" "context" $) }}
+  namespace: {{ .Release.Namespace }}
+{{- end -}}
+{{- end -}}
```
[the remark this answers is not in the document]
Comment (ahmednasir91) [reply bitnami/thanos/templates/querier/psp.yaml]: Done

File: bitnami/thanos/templates/querier/psp.yaml (hunk 2 above)
Comment (marcosbc) [reply bitnami/thanos/templates/querier/psp.yaml]: Thanks!

**Review by marcosbc — APPROVED** [review]
Looks good, thanks!

**Pull request merged**